### ISSUE TITLE
CLI bug

### ISSUE THREAD

**Issue by abel1502** [otakubeam/etude#1]
Whenever the CLI is run without any arguments, it crashes with the following message:
```
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string: construction from null is not valid
Aborted
```

(Built on kali-linux with clang++ 11.1.0-4, latest cmake, fmt and catch2)

**Comment by otakubeam** [comment]
Do you have the standard library installed? Because for me it **does** generate a proper error message about not being able to find `main.et`

Could you provide the backtrace? 

Like this (with systemd):
```
coredumpctl debug
... gdb starts
gdb> bt
```

**Comment by abel1502** [comment]
I'm using WSL2, so coredumpctl isn't available, but I just ran it with gdb like:
```
$ gdb ./etc
(gdb) run
```

Here's the backtrace:
```
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
#1  0x00007ffff74ca89f in __pthread_kill_internal (signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:78
#2  0x00007ffff747ea52 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff7469469 in __GI_abort () at ./stdlib/abort.c:79
#4  0x00007ffff77da909 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff77e5d8a in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff77e5df5 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77e6048 in __cxa_throw () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff77dd0d4 in std::__throw_logic_error(char const*) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x0000000000411ce1 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::allocator<char> > (this=this@entry=0x7fffffffcd38, __s=__s@entry=0x0, __a=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/basic_string.h:639
#10 0x00000000004116be in CompilationDriver::OpenFile[abi:cxx11](std::basic_string_view<char, std::char_traits<char> >) (this=<optimized out>, this@entry=0x7fffffffd898, name="main") at /mnt/d/Workspace/.MIPT/tmp/etude/src/driver/compil_driver.hpp:36
#11 0x000000000041095c in CompilationDriver::ParseOneModule (this=this@entry=0x7fffffffd898, name="main") at /mnt/d/Workspace/.MIPT/tmp/etude/src/driver/compil_driver.hpp:53
#12 0x000000000040fb9f in CompilationDriver::ParseAllModules (this=this@entry=0x7fffffffd898) at /mnt/d/Workspace/.MIPT/tmp/etude/src/driver/compil_driver.hpp:61
#13 0x00000000004091d1 in CompilationDriver::Compile (this=this@entry=0x7fffffffd898) at /mnt/d/Workspace/.MIPT/tmp/etude/src/driver/compil_driver.hpp:125
#14 0x0000000000408d63 in main (argc=<optimized out>, argv=<optimized out>) at /mnt/d/Workspace/.MIPT/tmp/etude/app/etc_cli.cpp:31
```

As for the standard library, I have now chacked, and it turns out I don't have the `ETUDE_STDLIB` env var set, but it had been in the working directory. With it explicitly, it does comlain adequately:
```
./stdlib
terminate called after throwing an instance of 'std::runtime_error'
  what():  Could not open file main
Aborted
```

**Comment by abel1502** [comment]
I should note, however, that the issue I had reported in telegram still persists even with the correct env var. I'll probably duplicate the report here as well:

Almost all tests fail, and I have tracked it down to qbe v1.1 not understanding the trailing comma in the construct `call $printf (l $strdata.0, ..., )`. The log message is `-:58: ) expected, got , instead`. It only seems to happen with the `...` in the arguments, as it doesn't complain about an earlier line with `call $cyclef3iPiPi ( w 40, l %.34, l %.35, )`.
